src/hazmat.rs

```diff
@@ -35,6 +35,10 @@ use curve25519_dalek::digest::{generic_array::typenum::U64, Digest};
 ///
 /// Instances of this secret are automatically overwritten with zeroes when they fall out of scope.
 pub struct ExpandedSecretKey {
+    // `scalar_bytes` and `scalar` are separate, because the public key is computed as an unreduced
+    // scalar multiplication (ie `mul_base_clamped`), whereas the signing operations are done
+    // modulo l.
+    pub(crate) scalar_bytes: [u8; 32],
```

Comment:
I guess I overlooked this before.

So, there are use cases like Ed25519-BIP32 where the private scalar is computed rather than clamped from bytes.

Adding this field means you can no longer construct an `ExpandedSecretKey` directly from a derived `Scalar` and prefix using the struct literal syntax `ExpandedSecretKey { scalar, hash_prefix }` and have to go through one of the constructor methods.

I guess the best you could do now if you have a `scalar` and a `hash_prefix` is to serialize the scalar and concatenate it with the `hash_prefix` and use `ExpandedSecretKey::from_bytes`?

Also all that aside, I find a mixture of public and private fields weird. If you really want to go down this path it would probably be good to make `scalar` and `hash_prefix` into methods, removing the fields from `pub` visibility.

Reply:
Opened a PR to make the other fields private: https://github.com/dalek-cryptography/curve25519-dalek/pull/544